modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManagerSelfTest.java

```diff
@@ -0,0 +1,880 @@
+                        throw new IgniteException(e);
+                    }
+                }
+            });
+
+        IgniteCache<?, ?> cache = ig.getOrCreateCache(DEFAULT_CACHE_NAME);
+
+        cache.destroy();
+
+        cpLatch.countDown();
+
+        snpFut.get(5_000, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * @param src Source node to calculate.
+     * @param grps Groups to collect owning parts.
+     * @param rmtNodeId Remote node id.
+     * @return Map of collected parts.
+     */
+    private static Map<Integer, Set<Integer>> owningParts(IgniteEx src, Set<Integer> grps, UUID rmtNodeId) {
+        Map<Integer, Set<Integer>> result = new HashMap<>();
+
+        for (Integer grpId : grps) {
+            Set<Integer> parts = src.context()
+                .cache()
+                .cacheGroup(grpId)
+                .topology()
+                .partitions(rmtNodeId)
+                .entrySet()
+                .stream()
+                .filter(p -> p.getValue() == GridDhtPartitionState.OWNING)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+
+            result.put(grpId, parts);
+        }
+
+        return result;
+    }
+
+    /**
+     * @param snpName Unique snapshot name.
+     * @param parts Collection of pairs group and appropratate cache partition to be snapshotted.
+     * @param snpSndr Sender which used for snapshot sub-task processing.
+     * @return Future which will be completed when snapshot is done.
+     */
+    private static IgniteInternalFuture<Boolean> startLocalSnapshotTask(
+        IgniteSnapshotManager snpMgr,
+        GridCacheDatabaseSharedManager dbMgr,
+        UUID srcNodeId,
+        String snpName,
+        Map<Integer, Set<Integer>> parts,
+        SnapshotSender snpSndr
+    ) throws IgniteCheckedException{
+        SnapshotFutureTask snpFutTask = snpMgr.registerSnapshotTask(snpName, srcNodeId, parts, snpSndr);
+
+        snpFutTask.start();
+
+        // Snapshot is still in the INIT state. beforeCheckpoint has been skipped
+        // due to checkpoint aready running and we need to schedule the next one
+        // right afther current will be completed.
+        dbMgr.forceCheckpoint(String.format(CP_TEST_SNAPSHOT_REASON, snpName));
+
+        snpFutTask.awaitStarted();
+
+        return snpFutTask;
+    }
+
+    /**
+     * @param ccfg Default cache configuration.
+     * @return Ignite instance.
+     * @throws Exception If fails.
+     */
+    private IgniteEx startGridWithCache(CacheConfiguration<Integer, Integer> ccfg, int range) throws Exception {
+        defaultCacheCfg = ccfg;
+
+        // Start grid node with data before each test.
+        IgniteEx ig = startGrid(0);
+
+        ig.cluster().baselineAutoAdjustEnabled(false);
+        ig.cluster().active(true);
+
+        for (int i = 0; i < range; i++)
+            ig.cache(DEFAULT_CACHE_NAME).put(i, i);
+
+        CheckpointProgress cpFut = ig.context()
```

Comment:
forceCheckpoint(ig)

Reply:
I should wait for the started checkpoint ends to ensure that all files are consistent on the disk. Currently, this method doesn't allow me to do so.

Reply:
Fixed.